### PR TITLE
fix(ch390): memory distribution error of phy_ctl1_reg_t; loopback not work properly (IDFGH-13804)

### PR DESCRIPTION
**Pull request by SergeyKharenko**
As mentioned in #37. 
Besides, add Version History in `README.md` to help users distinguish version differences.

### CODE REVIEW

File: ch390/src/esp_eth_phy_ch390.c [hunk 1]
```diff
@@ -41,9 +53,10 @@ typedef union {
     };
     uint32_t val;
 } phy_ctl1_reg_t;
```
Comment (kostaond) [inline ch390/src/esp_eth_phy_ch390.c:55]: Is this register description available only CH182**DS2** datasheet? I cannot find it in DS1 which is only available on their website? Is it possible to obtain CH182**DS2** datasheet somewhere?

Comment (SergeyKharenko) [reply ch390/src/esp_eth_phy_ch390.c:55]: Yes～I asked WCH technicians for ch182ds2 and they sent it to me via email and told me not to disclose its entire contents(it is okay to disclose the part, so I just wrote the definition of a register). But to be honest, I am really confused and worried. I wonder if there will be any legal risks😰😰😰~ If you really need this datasheet, you could either send an email to WCH on their official website, or send an email to me in private(addr is on my homepage, but please promise me not to let it spread).

Comment (kostaond) [reply ch390/src/esp_eth_phy_ch390.c:55]: OK, thank you for the info. I understand, it'll be better if you not share with me. I'll ask them by myself. Is it in English?

File: ch390/src/esp_eth_phy_ch390.c [hunk 2]
```diff
@@ -126,10 +139,10 @@ static esp_err_t ch390_loopback(esp_eth_phy_t *phy, bool enable)
 
     if (enable) {
         bmcr.en_loopback = 1;
-        phy_ctl1.pma_lpbk = 1;
+        phy_ctl1.remote_lpbk = 1;
```
Comment (kostaond) [inline ch390/src/esp_eth_phy_ch390.c:142]: Note to myself. Need to figure out, what kind of loopback it is. If it is far-end (as usually called by other vendors), it't not indented to be enabled here.

File: ch390/src/esp_eth_phy_ch390.c (hunk 1 above)
Comment (SergeyKharenko) [reply ch390/src/esp_eth_phy_ch390.c:55]: > OK, thank you for the info. I understand, it'll be better if you not share with me. I'll ask them by myself. Is it in English?

Nop~😢

Comment (kostaond) [reply ch390/src/esp_eth_phy_ch390.c:55]: I contacted WCH. They didn't share the datasheet but I consulted the matter with one of their representatives. Please see the comment below.

File: ch390/src/esp_eth_phy_ch390.c (hunk 2 above)
Comment (kostaond) [reply ch390/src/esp_eth_phy_ch390.c:142]: Reply from WCH:
"_IEEE802.3 standard interface MII register 0x0 (Control Register) bit 14 sets the loopback. PCS, PMA and remote belong to 4 different loopbacks. They belong to different layers. (more subdivided layers)

As mentioned above, different loopbacks are independent. You only need to enable one. It is recommended to use the loopback method disclosed in DS1, which is the loopback set by IEEE802.3 standard interface MII register 0x0 (Control Register) bit 14.

Other loopback methods are mainly used for hardware testing.

“Remote loopback” is the same as ”Far-end loopback“ mentioned by other vendors._"

So you don't need to enable it. It seems IEEE802.3 loopback only should be OK. You can use the common `phy_802_3` function.

Comment (kostaond) [reply ch390/src/esp_eth_phy_ch390.c:142]: @SergeyKharenko just a gentle reminder. Could you please resolve this comment based on feedback from WCH?

Comment (SergeyKharenko) [reply ch390/src/esp_eth_phy_ch390.c]: > @SergeyKharenko just a gentle reminder. Could you please resolve this comment based on feedback from WCH?

Sorry for my late reply~ I have been busy with my academic report recently😭
The commit has been submitted~ I fully delete the definition of phy_ctrl1_reg_t and its usage.

File: ch390/src/esp_eth_phy_ch390.c [hunk 3]
```diff
@@ -116,25 +93,16 @@ static esp_err_t ch390_loopback(esp_eth_phy_t *phy, bool enable)
     phy_802_3_t *phy_802_3 = esp_eth_phy_into_phy_802_3(phy);
```
Comment (kostaond) [inline ch390/src/esp_eth_phy_ch390.c:93]: I think we could delete `ch390_loopback` and use just `802_3` loopback function https://github.com/espressif/esp-idf/blob/master/components/esp_eth/src/phy/esp_eth_phy_802_3.c#L350

File: ch390/src/esp_eth_phy_ch390.c [hunk 4]
```diff
@@ -110,36 +87,6 @@ static esp_err_t ch390_get_link(esp_eth_phy_t *phy)
     return ret;
 }
```
Comment (bogdankolendovskyy) [inline ch390/src/esp_eth_phy_ch390.c:89]: @SergeyKharenko I've ran the loopback test from esp-idf's `components/esp_eth/test_apps`, and it seems like auto-negotiation is not supported when loopback is enabled.

The same happens with LAN8720, RTL8201, and DP83848, the same solution will work here. 

```suggestion

static esp_err_t ch390_autonego_ctrl(esp_eth_phy_t *phy, eth_phy_autoneg_cmd_t cmd, bool *autonego_en_stat)
{
    esp_err_t ret = ESP_OK;
    phy_802_3_t *phy_802_3 = esp_eth_phy_into_phy_802_3(phy);
    esp_eth_mediator_t *eth = phy_802_3->eth;
    if (cmd == ESP_ETH_PHY_AUTONEGO_EN) {
        bmcr_reg_t bmcr;
        ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, phy_802_3->addr, ETH_PHY_BMCR_REG_ADDR, &(bmcr.val)), err, TAG, "read BMCR failed");
        ESP_GOTO_ON_FALSE(bmcr.en_loopback == 0, ESP_ERR_INVALID_STATE, err, TAG, "Autonegotiation can't be enabled while in loopback operation");
    }
    return esp_eth_phy_802_3_autonego_ctrl(phy_802_3, cmd, autonego_en_stat);
err:
    return ret;
}

static esp_err_t ch390_loopback(esp_eth_phy_t *phy, bool enable)
{
    esp_err_t ret = ESP_OK;
    phy_802_3_t *phy_802_3 = esp_eth_phy_into_phy_802_3(phy);
    bool auto_nego_en = true;
    ESP_GOTO_ON_ERROR(ch390_autonego_ctrl(phy, ESP_ETH_PHY_AUTONEGO_G_STAT, &auto_nego_en), err, TAG, "get status of autonegotiation failed");
    ESP_GOTO_ON_FALSE(!(auto_nego_en && enable), ESP_ERR_INVALID_STATE, err, TAG, "Unable to set loopback while autonegotiation is enabled. Disable it to use loopback");
    return esp_eth_phy_802_3_loopback(phy_802_3, enable);
err:
    return ret;
}
```

Comment (SergeyKharenko) [reply ch390/src/esp_eth_phy_ch390.c:89]: @bogdankolendovskyy Thank you for your advice! Code will be updated tomorrow~

**Pull request merged**